talk/owt/sdk/p2p/p2pclient.cc

```diff
@@ -377,6 +377,10 @@ void P2PClient::OnMessageReceived(const std::string& remote_id,
                          remote_id, message);
 }
 void P2PClient::OnStopped(const std::string& remote_id) {
+  {
+    const std::lock_guard<std::mutex> removed_lock(removed_pc_mutex_);
+    removed_pc_ = pc_channels_[remote_id];
+  }
```

Comment:
We still have a possible race here: if there are two concurrent OnStopped() called. Say thread A starts first and replaces removed_pc. Then it loses the CPU on line 383. Then thread B goes through the whole function, replacing removed_pc. When thread A resumed, the pc_channels.erase(remote_id) will then trigger the deletion of the object while executing a callback, which we wanted to avoid. 

Solution:
acquire both locks at the start of the function, then do the other two operations holding both locks. removed_lock is only acquired in here, so no deadlock should be possible with other threads acquiring in reverse direction

To be extra safe, can follow what is done here: https://en.cppreference.com/w/cpp/thread/scoped_lock
notably:
```
        std::scoped_lock lock(e1.m, e2.m);
 
        // Equivalent code 1 (using std::lock and std::lock_guard)
        // std::lock(e1.m, e2.m);
        // std::lock_guard<std::mutex> lk1(e1.m, std::adopt_lock);
        // std::lock_guard<std::mutex> lk2(e2.m, std::adopt_lock);
```
scoped_lock requires C++17

Reply:
I noticed this as well, the original base framework acquired the locks in the same way, I just copied the structure. I am searching for which version of C++ the base framework is compiled, not sure it's 17.